Update SPEC file and enable more build options

aimylios opened the pull request:
This is a collection of some cosmetic changes to the SPEC file and to enable more build options.

I did not observe any issues in my tests, even with the Python scripting enabled. It should be fairly safe to merge this for the nightly builds, and if any users complain the new options can always be disabled again.

aimylios committed:
Update comment about generation of source files

The script "builder.sh" is now used to generate all required tarballs.

aimylios committed:
Set License tag to GPLv3+

KiCad is made available under the GNU General Public License (GPL)
version 3 or greater.
See http://kicad-pcb.org/about/licenses/ for details.

aimylios committed:
Update requirements

- replace bzr by git (not strictly required, but used to check the
  version number during build)
- sort BuildRequires and Requires tags

aimylios committed:
Use trailing '/' for all paths consistently

aimylios committed:
Update %files section

aimylios committed:
Enable KICAD_USE_OCE

OCE-visualization should be sufficient as runtime dependency, as it
will automatically pull in all other required OCE libraries.

aimylios committed:
Enable Python scripting

Only KICAD_SCRIPTING_WXPYTHON can not be set to ON because Fedora
currently does not provide a suitable wxPython package.
See https://bugzilla.redhat.com/show_bug.cgi?id=1519989 for details.

aimylios committed:
Remove kicad-4.0.0-nostrip.patch

It was added in 2014 without any testing to resolve a problem with an
empty kicad-debuginfo package. The first part of the patch was not
relevant for Linux anyways (MinGW only), and the debuginfo package is
just fine nowadays without the patch.
See https://bugzilla.redhat.com/show_bug.cgi?id=1076929 for details.

stevefalco commented:
I made a build using your patches, and I didn't see any problems.  From my perspective, these patches can be merged.

aimylios committed:
Clean up runtime dependencies

Remove Requires tags for compat-wxGTK3-gtk2, libcurl, OCE-visualization
and python2. RPM will take care of installing these (and some other
dependencies) automagically based on the information generated during
build.

aimylios committed:
Update description

Based on the Introduction chapter of KiCad's official documentation:
http://docs.kicad-pcb.org/master/en/kicad.html

aimylios commented:
I've just added two additional changes to update the description (as requested by @nickoe in issue #7) and to clean up the runtime dependencies. I am not sure whether the boost dependency could be removed as well. The `boost-devel` package is definitely required to build KiCad, but it seems to me that none of the boost libs is actually used at runtime. And even if there is one, it would be better to precisely require a specific package instead of the boost meta-package (which pulls in a lot of stuff that is definitely not required by KiCad).

nickoe commented:
This looks good to me. Thank you. Lets take it for a testrun.
